### PR TITLE
Fix string representation of devices

### LINKED ISSUE #632: __str__ and __repr__ incorrect in Device class
* *Expected behavior:* ``__str__`` and ``__repr__`` should produce useful strings

* *Actual behavior:* They produce strings where most entries are blank:
```
>>> dev = qml.device('default.qubit', 4)
>>> print(dev.__str__())
Default qubit PennyLane plugin
Name: 
API version: 
Plugin version: 
Author: 
>>>print(dev.__repr__())
pennylane.plugins.default_qubit.
Instance: 

```

* *Reproduces how often:* on every device that inherits from the base class without overwriting these methods

### PR DESCRIPTION
**Pull request by josh146**
**Context:** The existing `__str__()` and `__repr__()` methods of the abstract `Device` class were missing format parameters.

**Description of the Change:** Adds in string parameters.

Example:

```pycon
>>> import pennylane as qml
>>> dev1 = qml.device("qiskit.basicaer", wires=4)
>>> dev1
<BasicAerDevice (qiskit.basicaer, wires=4, shots=1024) at 0x7f05108a37f0>
>>> print(dev1)
Qiskit PennyLane plugin
Short name: qiskit.basicaer
Package: pennylane_qiskit
Plugin version: 0.9.0-dev
Author: Xanadu
Wires: 4
Shots: 1024
```

If a plugin developer wants to display additional plugin-specific information, they will need to manually define `__str__`. E.g., if the Qiskit device wants to display the provider information.

**Benefits:** Printing devices works as expected.

**Possible Drawbacks:** n/a

**Related GitHub Issues:** Fixes #632

### CODE REVIEW

File: pennylane/_device.py [hunk 1]
```diff
@@ -62,12 +62,20 @@ def __init__(self, wires=1, shots=1000):
 
     def __repr__(self):
         """String representation."""
-        return "{}.\nInstance: ".format(self.__module__, self.__class__.__name__, self.name)
+        return "<{} ({}, wires={}, shots={}) at {}>".format(
+            self.__class__.__name__, self.short_name, self.num_wires, self.shots, hex(id(self))
```
Comment (co9olguy) [inline pennylane/_device.py:66]: I think its unnecessary to have both name and short name in the less verbose representation.
e.g., this prints the following:
```
<DefaultQubit (default.qubit, wires=1, shots=1000) at 0x7fa9e2e3b590>
```

Comment (co9olguy) [inline pennylane/_device.py:66]: I always though "at 0xABD..." referred to the memory address. Turns out it doesn't!

Comment (co9olguy) [inline pennylane/_device.py:66]:
```suggestion
        return "<{} device (wires={}, shots={}) at {}>".format(
            self.__class__.__name__, self.num_wires, self.shots, hex(id(self))
```

Comment (josh146) [reply pennylane/_device.py:66]: Would you prefer to see the short name or the class name, if we had to pick one? e.g.,

```
<BasicAerDevice(wires=4, shots=1024) at 0x7f05108a37f0>
```

or 

```
<qiskit.basicaer(wires=4, shots=1024) at 0x7f05108a37f0>
```

Comment (josh146) [reply pennylane/_device.py:66]: Same here. I just used `id` originally, and was wondering why there was no `x`!

Comment (josh146) [reply pennylane/_device.py:66]: The class name looks more 'right' to me

Comment (josh146) [reply pennylane/_device.py:66]: Agree!

**Pull request merged**